src/Reducers/Auth/AuthReducer.js

```diff
@@ -0,0 +1,32 @@
+const authReducer = (state, action) => {
+  switch (action.type) {
+    case "SIGNUP":
+      return {
+        ...state,
+        user: action.payload.user,
+        token: action.payload.token,
+      };
+    case "LOGIN":
+      return {
+        ...state,
+        user: action.payload.user,
+        token: action.payload.token,
+      };
+    case "LOGOUT":
+      return {
+        ...state,
+        user: null,
+        token: null,
+      };
+    case "CHECKUSER":
```

Comment:
Here, three of your cases have the same return object. You can have a general case like `SET_USER` for `LOGIN`, `SIGNUP` and `CHECKUSER`.

Reply:
Or you can combine multiple cases together.

```js
switch(action.type) {
 case 'LOGIN':
 case 'SIGNUP': 
  return {
    ....
   }
}
```